.framework/source_guards.sh

```
@@ -0,0 +1,89 @@
+# Performance timing utilities for debugging
+if [[ -n "$DF_DEBUG_TIMING" ]]; then
+    _df_start_time=""
+
+    __df_start_timing() {
+        _df_start_time=$(date +%s%3N)
+    }
+
+    __df_end_timing() {
+        local label="$1"
+        if [[ -n "$_df_start_time" ]]; then
+            local end_time=$(date +%s%3N)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)